From eed43aadbbc8e52b90fe02a4d5d23624bae4cf9c Mon Sep 17 00:00:00 2001
From: Scott Tsai <scottt.tw@gmail.com>
Date: Wed, 26 Mar 2025 12:40:38 +0800
Subject: [PATCH] Disable solver/conv_wino_fury_RxS for gfx115x

conv_wino_fury_RxS produces the following error on gfx1151:

```
kernel: [drm:gfx_v11_0_bad_op_irq [amdgpu]] *ERROR* Illegal opcode in command stream
stderr: HSA_STATUS_ERROR_INVALID_ISA: The instruction set architecture is invalid. code: 0x100f
```

Test Program

```
import torch

device = 'cuda:0'
dtype = torch.float16
x = torch.zeros((1, 1, 3, 3)).to(device, dtype)
conv2d = torch.nn.Conv2d(1, 1, (3, 3)).to(device, dtype)
conv2d(x)
```

After this change, MIOpen would then choose the
`naive_conv_ab_nonpacked_fwd_nchw_half_double_half` kernel, which appears to work.
---
 src/solver/conv/conv_wino_fury_RxS.cpp | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/src/solver/conv/conv_wino_fury_RxS.cpp b/src/solver/conv/conv_wino_fury_RxS.cpp
index 6f90642fa..3be149cfd 100644
--- a/src/solver/conv/conv_wino_fury_RxS.cpp
+++ b/src/solver/conv/conv_wino_fury_RxS.cpp
@@ -251,6 +251,14 @@ bool ConvWinoFuryRxSCommon<Winodata, Winofilter>::IsApplicable(const ExecutionCo
         return false;
 
     const auto dev_name = ctx.GetStream().GetDeviceName();
+
+    if(StartsWith(dev_name, "gfx115")) {
+        // Triggers this error on gfx1151
+        // kernel: [drm:gfx_v11_0_bad_op_irq [amdgpu]] *ERROR* Illegal opcode in command stream
+        // stderr: HSA_STATUS_ERROR_INVALID_ISA: The instruction set architecture is invalid. code: 0x100f
+        return false;
+    }
+
     // All gfx11 ASICs are supported
     if(!StartsWith(dev_name, "gfx11"))
         return false;
-- 
2.49.0

